Source/HTTP/WinHttp/winhttp_provider.cpp

```diff
@@ -375,20 +400,26 @@ Result<HINTERNET> WinHttpProvider::GetHSession(uint32_t securityProtocolFlags)
         return hr;
     }
 
-    auto result = WinHttpSetOption(
-        hSession,
-        WINHTTP_OPTION_SECURE_PROTOCOLS,
-        &securityProtocolFlags,
-        sizeof(securityProtocolFlags));
-    if (!result)
+    // Only set secure protocols for HTTPS requests
+    // For HTTP requests, ignore the security protocol settings as they don't apply
+    if (isHttps)
     {
-        HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
-        HC_TRACE_ERROR_HR(HTTPCLIENT, hr, "WinHttpProvider WinHttpSetOption");
-        return hr;
+        auto result = WinHttpSetOption(
+            hSession,
+            WINHTTP_OPTION_SECURE_PROTOCOLS,
+            &securityProtocolFlags,
+            sizeof(securityProtocolFlags));
+        if (!result)
+        {
+            HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
+            HC_TRACE_ERROR_HR(HTTPCLIENT, hr, "WinHttpProvider WinHttpSetOption");
```

Comment:
not strictly related to this PR but it seems like this error could be clearer about the optinon it failed to set?

Reply:
yes, fixed.